pkg/minikube/bootstrapper/bsutil/kubeadm.go

```diff
@@ -63,6 +65,10 @@ func GenerateKubeadmYAML(cc config.ClusterConfig, n config.Node, r cruntime.Mana
 
 	cgroupDriver, err := r.CGroupDriver()
 	if err != nil {
+		// Whether it is a known failure and replace it with the correct ErrorType.
+		if known, err := IsKnownError(err); known {
```

Comment:
how about instead of parsing the error, if we get an a error we make a call to check if the service for docker is on

for example

if err != nil {

 //  check if docker service is on
/// if ! sysinit.New(runner).Active("docker");  { 
return ErrDockerRuntimeNotRunning
}

(and same for containerd)
}

Reply:
Yes, this solution is much better, and more general.

Do a code search, and I think maybe we can have a re-try logic in CGroupDrive()? If the operation fails, we use sysinit.New(r).Restart("docker") and re-try that operation?

Reply:
Find out that I can use sysinit.Manager.Active() function to check whether runtime is running. It is more general.
However, the only question is, Should I place this check after each "r.Runner.RunCmd()" for docker, containerd, and cri-o? Because we are not sure whether those command will fail too.